src/asImageWidthSrcSet.ts

```diff
@@ -0,0 +1,102 @@
+import { ImageFieldImage } from "@prismicio/types";
+import {
+	buildURL,
+	buildWidthSrcSet,
+	BuildWidthSrcSetParams,
+} from "imgix-url-builder";
+
+import { imageThumbnail as isImageThumbnailFilled } from "./isFilled";
+
+/**
+ * The return type of `asImageWidthSrcSet()`.
+ */
+type AsImageWidthSrcSetReturnType<Field extends ImageFieldImage> =
+	Field extends ImageFieldImage<"empty">
+		? null
+		: {
+				/**
+				 * The Image field's image URL with Imgix URL parameters (if given).
+				 */
+				src: string;
+
+				/**
+				 * A width-based `srcset` attribute value for the Image field's image
+				 * with Imgix URL parameters (if given).
+				 */
+				srcset: string;
+		  };
+
+/**
+ * Creates a width-based `srcset` from an Image field with optional image
+ * transformations (via Imgix URL parameters).
+ *
+ * If the Image field contains responsive views, each responsive view is used as
+ * a width in the resulting `srcset`.
+ *
+ * If a `widths` parameter is not given, `[400, 800, 1600]` will be used by default.
+ *
+ * @example
+ *
+ * ```ts
+ * const srcset = asImageWidthSrcSet(document.data.imageField, {
+ * 	widths: [400, 800, 1600],
+ * 	sat: -100,
+ * });
+ * // => {
+ * //   src:    'https://images.prismic.io/repo/image.png?sat=-100',
+ * //   srcset: 'https://images.prismic.io/repo/image.png?sat=-100&width=400 400w, ' +
+ * //           'https://images.prismic.io/repo/image.png?sat=-100&width=800 800w,' +
+ * //           'https://images.prismic.io/repo/image.png?sat=-100&width=1600 1600w'
+ * // }
+ * ```
+ *
+ * @param field - Image field (or one of its responsive views) from which to get
+ *   an image URL.
+ * @param params - An object of Imgix URL API parameters. The `widths` parameter
+ *   defines the resulting `srcset` widths.
+ *
+ * @returns A `srcset` attribute value for the Image field with Imgix URL
+ *   parameters (if given). If the Image field is empty, `null` is returned.
+ * @see Imgix URL parameters reference: https://docs.imgix.com/apis/rendering
+ */
+export const asImageWidthSrcSet = <Field extends ImageFieldImage>(
+	field: Field,
+	params: Omit<BuildWidthSrcSetParams, "widths"> &
+		Partial<Pick<BuildWidthSrcSetParams, "widths">> = {},
+): AsImageWidthSrcSetReturnType<Field> => {
+	if (isImageThumbnailFilled(field)) {
+		const { widths = [400, 800, 1600], ...urlParams } = params;
+		const {
+			url,
+			alt: _alt,
+			copyright: _copyright,
+			dimensions: _dimensions,
+			...responsiveViews
+		} = field;
+
+		// The Prismic Rest API will always return thumbnail values if
+		// the base size is filled.
+		const responsiveViewObjects: ImageFieldImage<"filled">[] =
+			Object.values(responsiveViews);
```

Comment:
Shouldn't the default view value be included in the matrix of `srcset`?

Like the default image is the desktop one, then we have a tablet and smartphone view?

Reply:
My bad, I don't know how to read

Reply:
You're right, this should have the source width on the default URL.

Reply:
Done!